src/client/components/forms/deletion.js

```diff
@@ -111,9 +111,9 @@ class EntityDeletionForm extends React.Component {
 								footer={footerComponent}
 								header={headerComponent}
 							>
-								If you’re sure that {entity.type} {entityName}
```

Comment:
So this gets transpiled to:

    'If you\u2019re sure that ',
    entity.type,
    ' ',
    entityName,
    'should be deleted, please enter a revision note below and confirm deletion.',

Given that, I think whitespace between `{} {}` is kept, but when a `{}` is at the end of the line, the leading whitespace on the line following that is ignored. So we just need to add a `{' '}` at any point where a variable substitution is currently at the end of the line (as is the case on line 114 here).

Reply:
This is very strange, I didn't look the transpiled code. The output on running with changes doesn't introduce double spaces - there is one space alright. 
Regarding insertion, this can be a good place: https://reactjs.org/blog/2014/02/20/react-v0.9.html#jsx-whitespace.

Reply:
Hmm I don't understand why there are no double spaces here. It might be best to make it `that {entity.type}{' '}` to avoid unwanted changes in the future.

Reply:
@shivam-tripathi I have an alternate proposition, adding a non-breaking space before should:
`If you’re sure that {entity.type} {entityName}
&nbsp;should be deleted, please enter a revision note
below and confirm deletion.`